plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts

```diff
@@ -296,23 +295,32 @@ export class GithubEntityProvider implements EntityProvider, EventSubscriber {
   /** {@inheritdoc @backstage/plugin-events-node#EventSubscriber.onEvent} */
   async onEvent(params: EventParams): Promise<void> {
     this.logger.debug(`Received event from ${params.topic}`);
-    if (params.topic !== TOPIC_REPO_PUSH) {
-      return;
-    }
+    if (EVENT_TOPICS.some(topic => topic === params.topic)) {
+      if (!this.connection) {
+        throw new Error('Not initialized');
+      }
 
-    await this.onRepoPush(params.eventPayload as PushEvent);
+      switch (params.topic) {
+        case 'github.push':
+          await this.onPush(params.eventPayload as PushEvent);
+          return;
+
+        case 'github.repository':
+          await this.onRepoChange(params.eventPayload as RepositoryEvent);
+          return;
+
+        default:
+          this.logger.error(`Missing case for event topic ${params.topic}`);
```

Comment:
is error really needed or would warning also do?

Reply:
In the end, it is a code path that should never be reached (see `EVENT_TOPICS`). It would signal that we have a topic that is accepted by the filter, subscribed, but not handled.

It is more an error for the package, not for the user's of the package.

There is a similar case at onRepoChange. However, I see that I changed it to/used `warn` there. I will align that. Thanks!